woocommerce/woocommerce-gutenberg-products-block
# Add an endpoint for getting pay for order orders

Add an endpoint for getting pay-for-order orders so we can use that to hydrate an order block in the checkout.





Fixes #8922 




#### Other Checks


### Screenshots
<img width="499" alt="Screen Shot 2023-06-28 at 6 39 15 PM" src="https://github.com/woocommerce/woocommerce-blocks/assets/56378160/09fae239-e73d-49d0-94bf-841e51c85bb7">

### Testing
1. Install and activate a plugin with pay-for-order flow. ex WooCommerce-Bookings
3. Create a product that needs confirmation
1. Purchase the product with pay-for-order flow
1. Go to WooCommerce -> Orders
1. Add some order fees for tax, shipping, etc
4. Click `Customer payment page` to go to the pay-for-order page
1. Copy the order_id and the key from the URL
1. Use Postman to test this endpoint `http://merchant.local/wp-json/wc/store/order/YOUR_ORDER_NUMBER?key=YOUR_KEY&billing_email=YOUR_BILLING_EMAIL`
2. Check that the order items, totals, ID, etc, are there

#### Automated Tests



### Changelog

> Add an endpoint for getting pay-for-order orders
